bioptim/dynamics/configure_problem.py

```diff
@@ -113,18 +128,54 @@ def torque_driven(ocp, nlp, with_contact: bool = False, fatigue: FatigueList = N
         ConfigureProblem.configure_qdot(nlp, True, False)
         ConfigureProblem.configure_tau(nlp, False, True, fatigue)
 
+        if implicit_dynamics:
+            ConfigureProblem.configure_qddot(nlp, False, True)
+            ocp.implicit_constraints.add(
+                ImplicitConstraintFcn.TAU_EQUALS_INVERSE_DYNAMICS,
+                node=Node.ALL_SHOOTING,
+                constraint_type=ConstraintType.IMPLICIT,
+            )
+        if nlp.model.nbSoftContacts() > 0:
+            if implicit_soft_contacts and not implicit_dynamics:
+                ConfigureProblem.configure_soft_contact_forces(nlp, False, True)
+            elif implicit_soft_contacts and implicit_dynamics:
+                raise ValueError(
+                    "This is not recommended to use both implicit_dynamics and implicit_soft_contacts."
+                    " Set one of them at False."
+                )
+
         if nlp.dynamics_type.dynamic_function:
             ConfigureProblem.configure_dynamics_function(ocp, nlp, DynamicsFunctions.custom)
         else:
             ConfigureProblem.configure_dynamics_function(
-                ocp, nlp, DynamicsFunctions.torque_driven, with_contact=with_contact, fatigue=fatigue
+                ocp,
+                nlp,
+                DynamicsFunctions.torque_driven,
+                with_contact=with_contact,
+                fatigue=fatigue,
+                implicit_dynamics=implicit_dynamics,
             )
 
         if with_contact:
             ConfigureProblem.configure_contact_function(ocp, nlp, DynamicsFunctions.forces_from_torque_driven)
 
+        ConfigureProblem.configure_soft_contact_function(ocp, nlp)
+
+        if nlp.model.nbSoftContacts() > 0 and implicit_soft_contacts and not implicit_dynamics:
+            ocp.implicit_constraints.add(
+                ImplicitConstraintFcn.SOFT_CONTACTS_EQUALS_SOFT_CONTACTS_DYNAMICS,
+                node=Node.ALL_SHOOTING,
+                constraint_type=ConstraintType.IMPLICIT,
+            )
```

Comment:
Please merge with the previous `if` if possible (either bring the other down or this one up)

Reply:
It is not possible to merge it, the order is of importance.